pkg/scanner/enrich_insightsv2.go

```diff
@@ -61,7 +61,32 @@ func (e *insightsBasedPackageEnricherV2) Enrich(pkg *models.Package,
 		return err
 	}
 
-	return e.applyInsights(pkg, res)
+	// Apply insights to the package
+	if err := e.applyInsights(pkg, res); err != nil {
+		logger.Debugf("Failed to apply insights to package: %s/%s: %v",
+			pkg.GetName(), pkg.GetVersion(), err)
+		return err
+	}
+
+	// For all dependencies call the callback function
+	for _, dep := range res.GetInsight().GetDependencies() {
+		if strings.EqualFold(dep.GetPackage().Name, pkg.GetName()) {
+			// Skip cyclic references
+			continue
+		}
+
+		if err := cb(&models.Package{
+			Manifest: pkg.Manifest,
+			Parent:   pkg,
+			Depth:    pkg.Depth + 1,
+			PackageDetails: models.NewPackageDetail(dep.GetPackage().GetEcosystem().String(),
```

Comment:
This will not work because we have mapping of ecosystem standard names. You probably need to use: https://github.com/safedep/vet/blob/main/pkg/models/models.go#L281